internal/overlord/servstate/handlers.go

```diff
@@ -162,11 +169,48 @@ func (m *ServiceManager) doStart(task *state.Task, tomb *tomb.Tomb) error {
 			}
 			releasePlan()
 		}
-		return fmt.Errorf("cannot start service: exited quickly with code %d", cmd.ProcessState.ExitCode())
+
+		logs, err := getLastLogs(logBuffer)
+		if err != nil {
+			logger.Noticef("Cannot read service %q logs: %v", req.Name, err)
+		}
+
+		msg := fmt.Sprintf("service exited too quickly with code %d", cmd.ProcessState.ExitCode())
+		if logs != "" {
+			msg += ":\n" + logs
+		}
+		return errors.New(msg)
 	}
 	// unreachable
 }
 
+// Used to strip the Pebble log prefix, for example: "2006-01-02T15:04:05.000Z [service] "
+// Timestamp must match format in logger.timestampFormat.
+var timestampServiceRegexp = regexp.MustCompile(`\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}\.\d{3}Z \[.*?\] `)
```

Comment:
Probably irrelevant here, but instead of `.*?` this is better spelled as `[^]]+`, which means "at least one character, but not ]". This prevents backtracking, as it can deterministically tell where to stop without depending on the rest of the string.

It also need to be anchored. More on this below.